governance/program/src/instruction.rs

```diff
@@ -0,0 +1,264 @@
+//! Program instructions
+
+use crate::state::{
+    Vote, MAX_GOVERNANCE_NAME_LENGTH, MAX_PROPOSAL_DESCRIPTION_LINK_LENGTH,
+    MAX_PROPOSAL_INSTRUCTION_DATA_LENGTH, MAX_PROPOSAL_NAME_LENGTH,
+};
+
+/// Instructions supported by the Governance program
+#[derive(Clone)]
+#[allow(clippy::large_enum_variant)]
+pub enum GovernanceInstruction {
+    /// Creates Governance account
+    ///
+    ///   0. `[writable]` Governance account. The account pubkey needs to be set to program-derived address (PDA) with the following seeds:
+    ///           1) 'governance' const prefix,
+    ///           2) Governed Program account key
+    ///   1. `[]` Account of the Program governed by this Governance account
+    ///   2. `[writable]` Program Data account of the Program governed by this Governance account
+    ///   3. `[signer]` Current Upgrade Authority account of the Program governed by this Governance account
+    ///   4. `[]` Governance mint that this Governance uses
+    ///   5. `[signer]` Payer
+    ///   6. `[]` System account
+    ///   7. `[]` Bpf_upgrade_loader account
+    ///   8. `[]` Council mint that this Governance uses [Optional]
+    CreateGovernance {
+        /// Vote threshold in % required to tip the vote
+        vote_threshold: u8,
+
+        /// Minimum slot time-distance from creation of proposal for an instruction to be placed
+        minimum_slot_waiting_period: u64,
+        /// Time limit in slots for proposal to be open to voting
+        time_limit: u64,
+        /// Governance name
+        name: [u8; MAX_GOVERNANCE_NAME_LENGTH],
```

Comment:
I meant the `name` field specifically.  Since the governance is created for one program, I don't know what additional useful information a "name" gives.

Reply:
Ok, thank you for clarification. 
The name field is used to provide a user friendly description for `Governance`. However given the discussion about specialisation https://github.com/solana-labs/solana-program-library/pull/1635#issuecomment-831502980 I think it should just be program_name unless there is a way to get program name given its address. Is the possible please?

Reply:
Hm, seems like we'll need another program just for program metadata in that case! there's no concept of a "name" for a program, except for hardcoded mappings in the explorer.  Considering we have `token-list`, perhaps we should also make a `program-list` or spinoff SPL Program Metadata from the Token Metadata program.  Either way, I would keep this field out of the governance.

Reply:
ok